dataland-batch-manager/src/main/kotlin/org/dataland/datalandbatchmanager/service/GleifGoldenCopyIngestor.kt

```diff
@@ -0,0 +1,132 @@
+    private val allCompaniesForceIngest: Boolean,
+
+    @Value("\${dataland.dataland-batch-managet.get-all-gleif-companies.flag-file:#{null}}")
+    private val allCompaniesIngestFlagFilePath: String?,
+) {
+    companion object {
+        const val WAIT_TIME: Long = 5000
+        const val UPLOAD_THREAT_POOL_SIZE = 32
+    }
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    /**
+     * Downloads the entire GLEIF golden copy file and uploads all included companies to the Dataland Backend.
+     * Does so only if the property "dataland.dataland-batch-managet.get-all-gleif-companies" is set.
+     */
+    @EventListener(ApplicationReadyEvent::class)
+    fun processFullGoldenCopyFileIfEnabled() {
+        val flagFile = allCompaniesIngestFlagFilePath?.let { File(it) }
+
+        if (allCompaniesForceIngest || flagFile?.exists() == true) {
+            if (flagFile?.exists() == true) {
+                logger.info("Found collect all companies flag. Deleting it.")
+                if (!flagFile.delete()) {
+                    logger.error(
+                        "Unable to delete flag file $flagFile. Manually remove it or import will " +
+                            "be triggered after service restart again.",
+                    )
+                }
+            }
+
+            logger.info("Retrieving all company data available via GLEIF.")
+            val tempFile = File.createTempFile("gleif_golden_copy", ".csv")
+            processFile(tempFile, gleifApiAccessor::getFullGoldenCopy)
+        } else {
+            logger.info("Flag file not present & no force update variable set => Not performing any download")
+        }
+    }
+
+    @Suppress("UnusedPrivateMember") // Detect does not recognise the scheduled execution of this function
+    @Scheduled(cron = "0 0 3 * * SUN")
+    private fun processDeltaFile() {
+        logger.info("Starting update cycle for latest delta file.")
+        val tempFile = File.createTempFile("gleif_update_delta", ".csv")
+        processFile(tempFile, gleifApiAccessor::getLastMonthGoldenCopyDelta)
+    }
+
+    @Synchronized
+    private fun processFile(csvFile: File, downloadFile: (file: File) -> Unit) {
+        waitForBackend()
+        val start = System.nanoTime()
+        try {
+            downloadFile(csvFile)
+            uploadCompanies(csvFile)
+        } finally {
+            if (!csvFile.delete()) {
+                logger.error("Unable to delete temporary file $csvFile")
+            }
+        }
+        logger.info("Finished processing of file $csvFile in ${getExecutionTime(start)}.")
+    }
+
+    private fun waitForBackend() {
+        while (true) {
+            try {
+                actuatorApi.health()
+                break
+            } catch (exception: ConnectException) {
+                logger.info("Waiting for 5s backend to be available. Exception was: ${exception.message}.")
+                Thread.sleep(WAIT_TIME)
+            }
+        }
+    }
+
```

Comment:
Macht es Sinn hier vllt eine MAX_TIME einzuführen?
Sonst läuft das ewig weiter im Falle eines Problems, oder?

Reply:
Finde ich so ok. Dann ist das backend nämlich irgendwie broken und es failt eh sonst alles.

Reply:
I took the arbitrary sufficiently large waiting time of 10 minutes.